tests/CppUTest/UtestTest.cpp

```diff
@@ -218,12 +222,26 @@ TEST(UtestShell, TestStopsAfterSetupFailure)
 }
 
 #if CPPUTEST_USE_STD_CPP_LIB
-static void thrownUnknownExceptionMethod_()
+#ifdef NEEDS_DISABLE_UNREACHABLE_CODE
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wunreachable-code"
```

Comment:
Ouch.

Could we wrap the throw in an "if" instead, which is always true in the test case? It wouldn't make the test much better but it would prevent us to have #defines and #pragma in the code.

Reply:
I'll give it a try.

Generally compilers are quite good in preventing these kind of issues now.